lib/syncthing/utils.go

```diff
@@ -124,3 +125,7 @@ func copyFile(src, dst string) error {
 func OpenGoleveldb(path string) (*db.Lowlevel, error) {
 	return db.Open(path)
 }
+
+func NewEventsLogger() *events.Logger {
```

Comment:
Why not just events.NewLogger(), Trying to keep the facade minimal?

Reply:
My principle was that you can do anything you need to run `syncthing.App` from `lib/syncthing`, i.e. you don't need to bind any other packages (which might e.g. not be go bindable).

Reply:
This is a tangent but I don't think we should aim for having a C-compatible interface to lib/syncthing. That forces un-Go-like choices (like passing in interfaces for dependencies). The people who want a C binding can create a small Go wrapper to do the required stuff.

Reply:
Why is passing interfaces un-Go-like? Honest question, as I thought that interfaces as function parameters are preferable to type, as it allows simpler testing by mocking (or if applicable replacing the type with a custom one).

And in this and earlier `lib/syncthing` PRs I don't remember having added any new interfaces for library purposes , just utility functions (the new event interface is due to the requested noop-logger). I believe there is something to be said for a library consumer not needing to bind multiple packages/libraries, if they don't use their functionality (events, database, config), but need the objects for the library initialisation. That's the purpose of `lib/syncthing/util.go`.

Reply:
Sorry no, I mean the exact opposite. Having the constructor accept interfaces for things like the logger would be very Go-ish, but not easily consumed by something like C I think.